src/server/context_test.ts

```diff
@@ -22,3 +26,26 @@ Deno.test("selectMiddlewares", () => {
   const mws = selectMiddlewares(url, mwRoutes);
   assert(mws.length === 4);
 });
+
+Deno.test({
+  name: "dynamic route conflicts",
+  fn() {
+    const routes = [
+      { name: "control-index", pattern: "/control" },
+      { name: "control-normal_route", pattern: "/control/normal_route" },
+      {
+        name: "two_dynamic-[second_dynamic]",
+        pattern: "/two_dynamic/:second_dynamic",
+      },
+      { name: "two_dynamic-[dynamic]", pattern: "/two_dynamic/:dynamic" },
+      { name: "override-[dynamic]", pattern: "/override/:dynamic" },
```

Comment:
I didn't add this exact one since I can't see how this could possibly cause an issue. I settled on this instead:
```ts
{ name: "nested2-[foo]/bar", pattern: "/nested2/:foo/bar" }, //routes/nested2/[foo]/bar.tsx
{ name: "nested2-foo/[bar]", pattern: "/nested2/foo/:bar" }, //routes/nested2/foo/[bar].tsx
```

Let me know if you want your exact case added.